app/main.py

```diff
@@ -1,2 +1,84 @@
+from typing import Hashable, Any, Optional, Iterable
+
+
 class Dictionary:
-    pass
+
+    def __init__(self) -> None:
+        self._capacity = 8
+        self._load_factor = 0.67
+        self._length = 0
+        self._hash_table: list = [None] * self._capacity
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self._length > self._capacity * self._load_factor:
+            self._increase_capacity()
+
+        index = self._find_index(key)
+        if not self._hash_table[index]:
+            self._length += 1
+        self._hash_table[index] = key, hash(key), value
+
+    def _increase_capacity(self) -> None:
+        table_to_resize = self._hash_table
+        self._capacity *= 2
+        self._hash_table = [None] * self._capacity
+        self._length = 0
+
+        for cell in table_to_resize:
+            if cell:
+                self.__setitem__(cell[0], cell[2])
+
+    def _find_index(self, key: Hashable) -> int:
+        index = hash(key) % self._capacity
+        while self._hash_table[index] and self._hash_table[index][0] != key:
+            index += 1
+            index %= self._capacity
+        return index
+
+    def __getitem__(self, key: Hashable) -> Any | KeyError:
```

Comment:
Do not indicate in the return type annotation errors that are raised throughout the method execution, [there is no such functionality](https://peps.python.org/pep-0484/#exceptions). Change it on line 64 as well.

Reply:
Thank you, I've already changed it.